pkg/handler/collector/deps_dev/deps_dev.go

```diff
@@ -123,6 +130,14 @@ func (d *depsCollector) populatePurls(ctx context.Context, docChannel chan<- *pr
 	if err != nil {
 		return fmt.Errorf("unable to retrieve datasource: %w", err)
 	}
+	start := time.Now()
+	err = d.getAllDependencies(ctx, ds.PurlDataSources)
+	if err != nil {
+		return fmt.Errorf("failed to get all dependencies: %w", err)
+	}
+	elapsed := time.Since(start)
+	// this is to know how long it takes to get all the dependencies for testing purposes
+	fmt.Printf("time taken to get all dependencies: %v \n", elapsed) // TODO: remove this when metrics are added
```

Comment:
You can only determine the time it takes to gather the necessary dependencies for testing in this way. 

The calls rely on channels. There's no other method to gauge it.

If better methods exist, I'd be glad to learn them and improve the code.

Reply:
Thanks.